addons/stock/models/product.py

```diff
@@ -1100,3 +1100,16 @@ def _adjust_uom_quantities(self, qty, quant_uom):
         else:
             computed_qty = self._compute_quantity(qty, procurement_uom, rounding_method='HALF-UP')
         return (computed_qty, procurement_uom)
+
+
+class ProductSupplierInfo(models.Model):
+    _inherit = 'product.supplierinfo'
+
+    delay = fields.Integer(inverse='_inverse_delay')
+
+    def _inverse_delay(self):
+        impacted_orderpoints = self.env['stock.warehouse.orderpoint'].search([
+            ('product_id', 'in', self.product_id.ids),
+        ])
```

Comment:
I think that the domain is too loose here. To me, you should not force a recompute of the orderpoints that are irrelevant. For instance all the order points that are not using a route related to suppliers are irrelevant here. Maybe an additional close like this would also be appropriate:
```diff
+ ("route_id", "=", "self.env.ref('purchase_stock.route_warehouse0_buy').id)"
```
Other routes might be relevant such as: dropship or subcontracting so that a pool of relevant route might need an update in certain modules..

Other criterions that I am not aware off might also be relevant to improve the perfs of this recomputation further.